In [1]:
import numpy as np
import scipy.stats as stats
import matplotlib.pyplot as plt

Cell 1: Import libraries and enter data

In [2]:
data1 = np.array([18.4, 19.1, 17.9, 18.7, 18.3])
data2 = np.array([20.2, 20.5, 20.1, 20.3, 20.4])

alpha = 0.05

print("Data 1:", data1)
print("Data 2:", data2)
print("Significance level:", alpha)

Data 1: [18.4 19.1 17.9 18.7 18.3]
Data 2: [20.2 20.5 20.1 20.3 20.4]
Significance level: 0.05


Cell 2: Calculate means, variances, and mean difference

In [3]:

n1 = len(data1)
n2 = len(data2)


mean1 = np.mean(data1)
mean2 = np.mean(data2)


var1 = np.var(data1, ddof=1)
var2 = np.var(data2, ddof=1)

mean_diff = mean1 - mean2

print("Mean of Data 1:", mean1)
print("Mean of Data 2:", mean2)

print("Variance of Data 1:", var1)
print("Variance of Data 2:", var2)

print("Mean Difference (Data 1 - Data 2):", mean_diff)

Mean of Data 1: 18.479999999999997
Mean of Data 2: 20.3
Variance of Data 1: 0.2020000000000008
Variance of Data 2: 0.024999999999999824
Mean Difference (Data 1 - Data 2): -1.8200000000000038


Cell 3: Perform Welch’s two-sample t-test

In [4]:

t_stat, p_value = stats.ttest_ind(
    data1,
    data2,
    equal_var=False
)

df = (
    (var1/n1 + var2/n2)**2
    /
    (
        (var1/n1)**2/(n1-1)
        +
        (var2/n2)**2/(n2-1)
    )
)

print("t-statistic:", t_stat)
print("Degrees of freedom:", df)
print("Two-tailed p-value:", p_value)

t-statistic: -8.54168323124357
Degrees of freedom: 4.975162325906963
Two-tailed p-value: 0.0003711295308672951


Cell 4: Calculate the 95% confidence interval

In [5]:

standard_error = np.sqrt(var1/n1 + var2/n2)


t_critical = stats.t.ppf(1 - alpha/2, df)


lower = mean_diff - t_critical * standard_error
upper = mean_diff + t_critical * standard_error

print("95% Confidence Interval for Mean Difference:")
print("Lower bound:", lower)
print("Upper bound:", upper)
print("CI:", (lower, upper))

95% Confidence Interval for Mean Difference:
Lower bound: -2.3685444300873075
Upper bound: -1.2714555699127001
CI: (np.float64(-2.3685444300873075), np.float64(-1.2714555699127001))


Cell 5: Decision and interpretation

In [6]:
print("Decision and Interpretation")

if p_value < alpha:
    print("Reject H0")
    print("There is a statistically significant difference")
    print("between the two population means.")
else:
    print("Fail to reject H0")
    print("There is not enough evidence of a significant difference.")

print("\nFinal Conclusion:")
print("The population means differ significantly at alpha = 0.05.")

Decision and Interpretation
Reject H0
There is a statistically significant difference
between the two population means.

Final Conclusion:
The population means differ significantly at alpha = 0.05.


Decision & Interpretation (plain English)

Since the p-value (0.0003711) is less than 0.05, we reject H₀. There is statistically significant evidence that the two population means differ.

The mean of Data 2 is higher than the mean of Data 1 by 1.82 units in the sample. The confidence interval for Data 1 − Data 2 is entirely negative, which is consistent with this difference.

Note: Your theory also covers the pooled t-test. If your teacher specifically asks for the equal-variance (pooled) test, the result is t = −8.5417, df = 8, and p ≈ 0.0000272; the decision is still to reject H₀.